crates/bevy_render/src/render_resource/shader.rs

```diff
@@ -143,19 +144,35 @@ impl ProcessedShader {
                 },
             )?,
         };
-        let module_info = naga::valid::Validator::new(
-            naga::valid::ValidationFlags::default(),
-            naga::valid::Capabilities::default(),
-        )
-        .validate(&module)?;
+        let mut capabilities = naga::valid::Capabilities::empty();
+        // iterate over all capabilities
+        let all_bits: u8 = naga::valid::Capabilities::all().bits();
+        for i in 0..8 - all_bits.leading_zeros() {
```

Comment:
This is more complex than it was before. Does the following not work?
```suggestion
        const PAIRS: &[(wgpu::Feature, naga::valid::Capabiliites)] = &[(..., ...), ...];
        for (feature, capability) in PAIRS {
```

Reply:
The advantage of my solution is that it fails with "unimplemented" if new Capabilities are added and they are not addressed in the match. But your solution is obviously better if readability is important.

Reply:
Maybe your solution plus some `assert_eq!(naga::valid::Capabilites::all(), ...)`?

Reply:
I'd prefer for there to be less runtime cost and readability over it failing here. This complexity isn't worth it.